client/java/src/main/java/io/openlineage/client/transports/HttpTransport.java

```diff
@@ -117,7 +117,11 @@ public void emit(@NonNull OpenLineage.RunEvent runEvent) {
   private void throwOnHttpError(@NonNull HttpResponse response) throws IOException {
     final int code = response.getStatusLine().getStatusCode();
     if (code >= 400 && code < 600) { // non-2xx
-      throw new OpenLineageClientException(EntityUtils.toString(response.getEntity(), UTF_8));
+      String message =
+          String.format(
+              "code: %d, response: %s", code, EntityUtils.toString(response.getEntity(), UTF_8));
```

Comment:
this includes http code in a log which is helpful

Reply:
This is a very nice quality of life improvement!